lib/ansible/modules/cloud/azure/azure_rm_webappslot.py

```diff
@@ -0,0 +1,1039 @@
+                                                 "java_version",
+                                                 "php_version",
+                                                 "python_version",
+                                                 "linux_fx_version"]
+
+        self.supported_linux_frameworks = ['ruby', 'php', 'dotnetcore', 'node', 'java']
+        self.supported_windows_frameworks = ['net_framework', 'php', 'python', 'node', 'java']
+
+        super(AzureRMWebAppSlots, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                                 mutually_exclusive=mutually_exclusive,
+                                                 supports_check_mode=True,
+                                                 supports_tags=True)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            if hasattr(self, key):
+                setattr(self, key, kwargs[key])
+            elif kwargs[key] is not None:
+                if key == "scm_type":
+                    self.site_config[key] = kwargs[key]
+
+        old_response = None
+        response = None
+        to_be_updated = False
+
+        # set location
+        resource_group = self.get_resource_group(self.resource_group)
+        if not self.location:
+            self.location = resource_group.location
+
+        # get web app
+        webapp_response = self.get_webapp()
+
+        if not webapp_response:
+            self.fail("Web app {0} does not exist in resource grouop {1}.".format(self.webapp_name, self.resource_group))
+
+        # get slot
+        old_response = self.get_slot()
+
+        # set is_linux
+        is_linux = True if webapp_response['reserved'] else False
+
+        if self.state == 'present':
+            if self.frameworks:
+                # java is mutually exclusive with other frameworks
+                if len(self.frameworks) > 1 and any(f['name'] == 'java' for f in self.frameworks):
+                    self.fail('Java is mutually exclusive with other frameworks.')
+
+                if is_linux:
+                    if len(self.frameworks) != 1:
+                        self.fail('Can specify one framework only for Linux web app.')
+
+                    if self.frameworks[0]['name'] not in self.supported_linux_frameworks:
+                        self.fail('Unsupported framework {0} for Linux web app.'.format(self.frameworks[0]['name']))
+
+                    self.site_config['linux_fx_version'] = (self.frameworks[0]['name'] + '|' + self.frameworks[0]['version']).upper()
+
+                    if self.frameworks[0]['name'] == 'java':
+                        if self.frameworks[0]['version'] != '8':
+                            self.fail("Linux web app only supports java 8.")
+                        if self.frameworks[0]['settings'] and self.frameworks[0]['settings']['java_container'].lower() != 'tomcat':
```

Comment:
framwork_spec doesn't look like it requires `settings`, so potentially KeyErrors in several places here if not specified (probably should use `.get('settings', {})` or something similar, depending on usage.

Reply:
fixed